### ISSUE TITLE
isBefore() throws Error when input is not a Date

### ISSUE THREAD

**Issue by caiocmpaes** [ctavan/express-validator#174]
When *isBefore()* receives a **string** that is not a Date It is throwing an Error (**Not the expected validation error**).

Example:
```
// req.body.birthdate == 'str'
req.checkBody('birthdate').isAlpha(); // No validation error 
req.checkBody('birthdate').isDate(); // => Add error to req.validationErrors()...
req.checkBody('birthdate').isBefore(new Date()); // throws Type Error
```

Stack trace:
```
Cannot read property 'then' of null
TypeError: Cannot read property 'then' of null
    at ValidatorChain.isBefore (/...mypath.../node_modules/express-validator/lib/express_validator.js:274:16)
    at /...mypath.../routes/routes.js:20:49
```

Is this the expected behavior for the isBefore()? Or It should have just add one more error to the queue?

**Comment by rustybailey** [comment]
This seems to have something to do the way `isBefore()` works in [node-validator](https://github.com/chriso/validator.js). I have opened an issue here: https://github.com/chriso/validator.js/issues/435.

As a workaround in the meantime, I would have `var validator = require('express-validator').validator;` at the top of your file and change that last line to
```
if (validator.isDate(req.body.birthdate)) {
    req.checkBody('birthdate').isBefore(new Date());
}
```
That way you're only doing a comparison check if the 'birthdate' is a valid date. Please let me know if you have any questions.

**Comment by caiocmpaes** [comment]
Thanks, @rustybailey . Your workaround is what I need now.

Before opening this issue I checked the validator module and have seen that `validator.isBefore('str', new Date())` returns `null`. Maybe this is the root cause of the problem. For me, the validations should always return `true` or `false`.

**Comment by rustybailey** [comment]
Agreed. We'll see what their response is.

**Comment by rustybailey** [comment]
I opened [this PR](https://github.com/chriso/validator.js/issues/435), so once it is merged in and a new node-validator version is pushed, I just need to update the validator version in `package.json` and publish a new version of express-validator.

**Comment by caiocmpaes** [comment]
Just saw your PR on node-validator. Lets wait. Thanks!

**Comment by rustybailey** [comment]
I just remembered we changed our validator version to 4.0.x, and the fix is in 4.0.6. Just to be safe, I would delete your `node_modules` folder, do an `npm install` and you should be good.
